api/db/migrations/20201102143300_migrate-samlid-to-authentication-methods.js

```diff
@@ -0,0 +1,44 @@
+const { batch } = require('../batchTreatment');
+const AuthenticationMethod = require('../../lib/domain/models/AuthenticationMethod');
+
+const TABLE_AUTHENTICATION_METHODS = 'authentication-methods';
+const TABLE_USERS = 'users';
+const SAML_ID_COLUMN = 'samlId';
+
+exports.up = async function(knex) {
+
+  const usersWithSamlId = await knex(TABLE_USERS).select('id', SAML_ID_COLUMN).whereNotNull(SAML_ID_COLUMN);
+
+  await batch(knex, usersWithSamlId, (userWithSamlId) => {
```

Comment:
Est-ce que ces deux instructions s'exécutent dans la même transaction ?
Si ce n'est pas le cas, risque de compte GAR créé pendant la migration dans `users.samlId` et pas dans `authentication-methods`

```suggestion
  const usersWithSamlId = await knex(TABLE_USERS).select('id', SAML_ID_COLUMN).whereNotNull(SAML_ID_COLUMN);

  await batch(knex, usersWithSamlId, (userWithSamlId) => {
```

Reply:
Je viens de vérifier est apparemment les requêtes sont bien exécutées dans la même transaction:
`  knex:query BEGIN; trx2 +6ms
  knex:query update "knex_migrations_lock" set "is_locked" = ? where "is_locked" = ? trx2 +1ms
  knex:query select * from information_schema.tables where table_name = ? and table_schema = current_schema() trx2 +1ms
  knex:query select * from information_schema.tables where table_name = ? and table_schema = current_schema() trx2 +1ms
  knex:query select * from "knex_migrations_lock" trx2 +2ms
  knex:query select "name" from "knex_migrations" order by "id" asc trx2 +0ms
  knex:query select max("batch") as "max_batch" from "knex_migrations" trx2 +2ms
  knex:query alter table "stages" alter column "message" drop default trx2 +2ms
  knex:query alter table "stages" alter column "message" drop not null trx2 +0ms
  knex:query alter table "stages" alter column "message" type text using ("message"::text) trx2 +1ms
  knex:query insert into "knex_migrations" ("batch", "migration_time", "name") values (?, ?, ?) trx2 +1ms
  knex:query update "knex_migrations_lock" set "is_locked" = ? trx2 +1ms
  knex:query COMMIT; trx2 +1ms
`